sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -3315,6 +3329,54 @@ class Analyzer(override val catalogManager: CatalogManager) extends RuleExecutor
     }
   }
 
+  /**
+   * Resolves columns of an output table from the data in a logical plan. This rule will:
+   *
+   * - Insert casts when data types do not match
+   * - Detect plans that are not compatible with the output table and throw AnalysisException
+   */
+  object ResolveSetVariable extends Rule[LogicalPlan] {
```

Comment:
once code appears in `Analyzer.scala`, we generally don't ever move it out (to avoid messing up the GitHub blame history). However `Analyzer.scala` is already huge with many thousands of lines; could we move this new rule into a separate file instead if we can?

Reply:
+1